test/export/note_sorter_spec.py

```diff
@@ -0,0 +1,148 @@
+import logging
+
+from mamba import describe, it, context
+from unittest.mock import MagicMock
+from random import shuffle
+
+from test_utils.anki import mock_anki_modules
+
+mock_anki_modules()
+
+from crowd_anki.export.note_sorter import NoteSorter
+from crowd_anki.config.config_settings import ConfigSettings, NoteSortingMethods
+
+test_guids = ["abc", "bcd", "cde", "def", "efg", "fgh"]
+test_flags = [0, 1, 2, 3, 4, 5]
+test_tags = ["adjectives", "directions", "interesting", "nouns", "verbs", "zzzzFinal"]
+test_notemodels = ["Default", "LL Noun", "LL Sentence", "LL Verb", "LL Word", "Zulu"]
+test_notemodelids = test_guids
+test_fields = test_tags
+
+note_sorting_single_result_pairs = [
+    (NoteSortingMethods.GUID, test_guids),
+    (NoteSortingMethods.FLAG, test_flags),
+    (NoteSortingMethods.TAG, test_tags),
+    (NoteSortingMethods.NOTE_MODEL, test_notemodels),
+    (NoteSortingMethods.NOTE_MODEL_ID, test_notemodelids),
+    (NoteSortingMethods.FIELD1, test_fields),
+    (NoteSortingMethods.FIELD2, test_fields)
+]
+
+test_multikey_notemodel_guid = [(notemodel, guid) for notemodel in test_notemodels for guid in test_guids]
```

Comment:
this seems to create `N^2` number of pairs, did you want something like `zip` instead?